app/src/components/Header/Header.tsx

```diff
@@ -0,0 +1,98 @@
+import { useState, useEffect, useContext } from "react";
+import ModalMenuContext from "@context/ModalMenu/ModalMenu";
+import "./Header.scss";
+import "@components/Header/Header.scss";
+import Button from "@components/ui/Button/Button";
+import Logo from "@assets/logo4.svg?react";
+import Ellipsis from "@assets/icons/ellipsis.svg?react";
+import Plus from "@assets/icons/plus.svg?react";
+import ChevronDown from "@assets/icons/chevron.svg?react";
+import BREAKPOINTS from "@utils/Consts/breakpoints";
+
+const Header = () => {
+  const modalMenuCtx = useContext(ModalMenuContext);
+
+  const [isVisible, setIsVisible] = useState<boolean>(false);
+
+  const mql = window.matchMedia(`(max-width: ${BREAKPOINTS["screen-md-max"]})`);
+  const [matches, setMatches] = useState(mql.matches);
+
+  useEffect(() => {
+    const handleChangeScreenSize = (event: MediaQueryListEvent) => {
+      setMatches(event.matches);
+
+      if (!event.matches) {
+        modalMenuCtx.closeModalMenu();
+      }
+    };
+    mql.addEventListener("change", handleChangeScreenSize);
+
+    return () => {
+      mql.removeEventListener("change", handleChangeScreenSize);
+    };
+  }, [mql, modalMenuCtx]);
+
+  const handleToggleButtonMenu = () => {
+    if (isVisible) {
+      setIsVisible(false);
+    } else {
+      setIsVisible(true);
+    }
+  };
+
+  const renderButton = () => {
+    return <Button Icon={<Plus />} variant="secondary" />;
+  };
+  const handleToggleDropdown = () => {
+    if (modalMenuCtx.isModalMenuOpen) {
+      modalMenuCtx.closeModalMenu();
+    } else {
+      modalMenuCtx.openModalMenu();
```

Comment:
if isModalMenuOpen is true - then nothing should happen, right? you shouldn't open it once again